src/main/java/seedu/us/among/ui/ResultDisplay.java

```diff
@@ -144,10 +144,65 @@ public String getColorCode(String statusCode) {
      */
     public void feedbackHelper(String feedbackToUser) {
+        String[] specialFeedback = {"New API", "Edited", "Method"};
         responseMeta.getChildren().clear();
+        boolean match = false;
+        for (String special : specialFeedback) {
+            match = feedbackToUser.startsWith(special);
+            if (match) {
+                break;
+            }
+        }
+        if (match) {
+            int methodStart = feedbackToUser.indexOf("Method") + 8;
+            String cutString = feedbackToUser.substring(methodStart);
+            int methodEnd = cutString.indexOf("\n") + methodStart;
+            String methodName = feedbackToUser.substring(methodStart, methodEnd);
+            String noData = "No Data";
+            Label method = new Label(String.format("Method: %s", methodName));
+            Label statusCodeAndPhrase = new Label(String.format("Status: %s", noData));
+            Label responseTime = new Label(String.format("Time: %s", noData));
+            Label protocol = new Label(String.format("Protocol: %s", noData));
+            String colorCode = "-fx-background-color: #999999";
+            method.setStyle(colorCode);
+            statusCodeAndPhrase.setStyle(colorCode);
+            responseTime.setStyle(colorCode);
+            protocol.setStyle(colorCode);
+            responseMeta.getChildren().addAll(method, statusCodeAndPhrase, responseTime, protocol);
+            //beautifying feedback
+            String beautified = beautify(feedbackToUser);
+            feedbackToUser = beautified;
+        }
         resultDisplay.setText(feedbackToUser);
     }
 
+    private String beautify(String feedback) {
+        String dividers = "============\n\n";
+        int methodStart = feedback.indexOf("Method");
+        String result = feedback.substring(0, methodStart) + "Endpoint:\n" + dividers;
+        int addressStart = feedback.indexOf("http");
+        String address = feedback.substring(addressStart);
+        int addressEnd = address.indexOf("\n") + addressStart;
+        result = result + feedback.substring(addressStart, addressEnd) + "\n\n";
+        int headerStart = feedback.indexOf("Headers:\n") + 9;
+        int tagStart = feedback.indexOf("Tags:\n");
+        int meta = feedback.indexOf("Last Response:");
+        int response = feedback.indexOf("Response Body:\n");
+        if (!(headerStart < 9)) {
+            result = result + "Headers:\n" + dividers;
+            if (!(tagStart < 7)) {
+                result = result + feedback.substring(headerStart, tagStart);
+            } else {
+                result = result + feedback.substring(headerStart, meta);
+            }
+        }
+        if (tagStart > 0) {
+            result = result + "Tags:\n" + dividers + feedback.substring(tagStart + 6, meta);
+        }
+        result = result + "Last Saved Response:\n" + dividers + feedback.substring(response + 15);
+        return result;
+    }
+
```

Comment:
this beatify implementation does not hold if we change the toString output of feedback, endpoint or anything related to these comments. Hence I will have to reject this implementation of beautify.